src/Servers/Kestrel/Transport.Quic/src/WebHostBuilderQuicExtensions.cs

```diff
@@ -20,15 +19,10 @@ public static class WebHostBuilderQuicExtensions
     /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
     public static IWebHostBuilder UseQuic(this IWebHostBuilder hostBuilder)
     {
-        if (QuicListener.IsSupported)
+        return hostBuilder.ConfigureServices(services =>
```

Comment:
The presence of a registered multiplexed transport factory is tested when figuring out if an endpoint supports HTTP/3. For example: https://github.com/dotnet/aspnetcore/blob/9de8e33d4d3feb7c53b7a92af2b56885e8b4c603/src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs#L180

Reply:
Good point. This isn't relevant if you have HTTP/3 disabled which is the .NET 7 default. Or if you have only HTTP/3 enabled which would throw either way.

But if you have `HttpProtocols.Http1AndHttp2AndHttp3` configured which is the default in `main`, we have to pay the cost of the check regardless 😢. Do you think we could leverage your new `CanBind` API (#44537) for checking this in .NET 8?

If we backport this to 7 (which I still think we should do), we might just have to swallow the NotSupportedException thrown from BindAsync in the Http1AndHttp2AndHttp3 case. Fortunately, that's not the default in 7, and I don't see many people manually setting that unless they are running on

Reply:
I think we should ask the S.N.Q team to look at improving `QuicListener.IsSupported` performance. See if it can be changed to avoid loading msquic.

Reply:
> This isn't relevant if you have HTTP/3 disabled which is the .NET 7 default

https://github.com/dotnet/runtime/pull/73153 enabled HTTP/3 by default for 7.0 rc1.

Reply:
> [dotnet/runtime#73153](https://github.com/dotnet/runtime/pull/73153) enabled HTTP/3 for HttpClient by default for 7.0 rc1.

The developer still has to opt-in by changing the request version or policy.